Makefile

```diff
@@ -25,40 +25,29 @@ export DEFAULT_ROUTING ?= basic
 export KUBECONFIG ?= ${HOME}/.kube/config
 export DEVWORKSPACE_API_VERSION ?= 1f335562c475972132851c68227dd36558317bb3
 
+# Set GO_PACKAGE_PATH to what is found in go.mod
+GO_PACKAGE_PATH ?= $(shell head -n1 go.mod | cut -d " " -f 2)
+
+# Set BUILD_TIME
+BUILD_TIME = $(shell date -u '+%Y-%m-%dT%H:%M:%SZ')
+
+# Set GIT_COMMIT_ID
+REF = $(shell cat .git/HEAD | sed 's|ref: ||')
+GIT_COMMIT_ID := $(shell cat .git/${REF})
+
 #internal params
 DEVWORKSPACE_CTRL_SA=devworkspace-controller-serviceaccount
 INTERNAL_TMP_DIR=/tmp/devworkspace-controller
 BUMPED_KUBECONFIG=$(INTERNAL_TMP_DIR)/kubeconfig
 RELATED_IMAGES_FILE=$(INTERNAL_TMP_DIR)/environment
 
-ifeq (,$(shell which kubectl))
-ifeq (,$(shell which oc))
-$(error oc or kubectl is required to proceed)
-else
-K8S_CLI := oc
-endif
-else
-K8S_CLI := kubectl
-endif
-
-ifeq ($(shell $(K8S_CLI) api-resources --api-group='route.openshift.io'  2>&1 | grep -o routes),routes)
-PLATFORM := openshift
-else
-PLATFORM := kubernetes
-endif
-
-# minikube handling
-ifeq ($(shell $(K8S_CLI) config current-context 2>&1),minikube)
-# check ingress addon is enabled
-ifeq ($(shell minikube addons list -o json | jq -r .ingress.Status), disabled)
-$(error ingress addon should be enabled on top of minikube)
-endif
-export ROUTING_SUFFIX := $(shell minikube ip).nip.io
+ifneq (,$(shell which kubectl)$(shell which oc))
```

Comment:
I'm not sure I fully got what you try to achieve with it.

Reply:
This was to avoid errors around kubetclt/oc not being present. in particular this: 

```
ifeq (,$(shell which kubectl))
ifeq (,$(shell which oc))
$(error oc or kubectl is required to proceed)
```

deploy.mk should only be included when oc and/or kubectl is present

Reply:
The issue is that other rules will fail for a more cryptic reason if `kubectl/oc` aren't installed, e.g.
```
make run
Current env vars:
    NAMESPACE=devworkspace-controller
    DWO_IMG=quay.io/devfile/devworkspace-controller:next
    PULL_POLICY=Always
    ROUTING_SUFFIX=192.168.99.100.nip.io
    DEFAULT_ROUTING=basic
    DEVWORKSPACE_API_VERSION=1f335562c475972132851c68227dd36558317bb3
Versions:
    GIT_COMMIT_ID=819de03bd85fa9c6ea67596f575e5a0bc36b7eb3
    BUILD_TIME=2021-04-27T19:05:33Z
make: *** No rule to make target '_gen_configuration_env', needed by 'run'.  Stop.
```

Reply:
@amisevsk Yeah, that makes sense.. except since we are calling these new targets from the docker file it doesn't have oc/kubectl so we will always see this error if we always include it... hmm.. maybe there is another way around this though?

Reply:
I've moved the other related targets to the deploy.mk - sorry for not being more clear :woman_facepalming: